R/autoxgboost.R

```diff
@@ -104,11 +106,21 @@ autoxgboost = function(task, measure = NULL, control = NULL, par.set = NULL, max
   }
 
   if (has.cat.feats) {
-    if (factor.encoder == "impact") {
-      base.learner = makeImpactFeaturesWrapper(base.learner)
-    } else {
-      task = createDummyFeatures(task)
+    d = getTaskData(task)
+    d = dropNamed(d, getTaskTargetNames(task))
+    feat.cols = colnames(d)[vlapply(d, is.factor)]
+    impact.cols = colnames(d[, vlapply(d, function(x) is.factor(x) && nlevels(x) > impact.encoding.boundary)])
+    dummy.cols = setdiff(feat.cols, impact.cols)
+    if (length(dummy.cols) != 0L)
+      base.learner = makeDummyFeaturesWrapper(base.learner, cols = dummy.cols)
+    if (length(impact.cols) != 0L) {
+      base.learner = makeImpactFeaturesWrapper(base.learner, cols = impact.cols)
+      par.set$pars$slope.param = makeIntegerParam("slope.param", lower = 0L, upper = 2^12)
```

Comment:
you can use the `c()` operator to join param sets

Reply:
I created a ````impactencodingparset```` which will be joint with ````par.set```` if impact encoding is used.